### PR TITLE
[3.1.0] Update Primary User Stores Docs

### PR DESCRIPTION
**Pull request by Sarangan0219**
## Purpose

- Fixes https://github.com/wso2/docs-apim/issues/1946 and https://github.com/wso2/docs-apim/issues/2070
- Adds Missing configurations
- Updates outdated configurations

## Affected products

APIM 3.1.0

### CODE REVIEW

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 # Configuring a JDBC User Store
 
-JDBC user stores can be configured using the <APIM_HOME>/repository/conf/deployment.toml file's [user_store] configuration section. The user core connects to two databases (the same database is utilized by default):
+User management functionality is provided by default in WSO2 API Manager and it can be configured in the 
+`<API-M_HOME>/repository/conf/deployment.toml` file. The changes done in the `deployment.toml` file will be automatically populated to the `<API-M_HOME>/repository/conf/user-mgt.xml` file as well.
```
Comment (vithu30) [inline en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md:4]: IMO, we do not need mention about the old configuration file - `user-mgt.xml`. Shall we only mention about deployment.toml?

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md [hunk 2]
```diff
@@ -13,11 +22,18 @@ You can either use the default configuration or you can change it in the followi
...
+-   [Step 3: Updating the datasources](#step-3-updating-the-datasources)
+-   [Step 4: Starting the server](#step-4-starting-the-server)
+
 ### Step 1: Configuring the JDBC user store manager
 
-1. To configure a JDBC user store as the primary user store, you must give the value: `database` for the config: `user_store.type` in the `<APIM_HOME>/repository/conf/deployment.toml` file.
+-   Add these configurations below in `<API-M_HOME>/repository/conf/deployment.toml` file.
 
-    ```toml
+    ```
```
Comment (vithu30) [inline en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md:36]: Did we test configuring the userstore type as `database`. AFAIR, there were some issues when setting the userstore type as `database`. We need to set it as `database_unique_id` to resolve it.  could you please double check on this? :)

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md (hunk 1 above)
Comment (Sarangan0219) [reply en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md]: Removed the sections about the old configuration file `user-mgt.xml`  and updated the section to use this config to verify the configs are correctly populated or not.

https://github.com/wso2/docs-apim/pull/2174/commits/c21f3d0bd012d6cbf0be75733e110326ae5e53f3

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md (hunk 2 above)
Comment (Sarangan0219) [reply en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-jdbc-user-store.md:33]: Corrected https://github.com/wso2/docs-apim/pull/2174/commits/c21f3d0bd012d6cbf0be75733e110326ae5e53f3

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-the-primary-user-store.md [hunk 3]
```diff
@@ -26,32 +24,37 @@ Instead of using the embedded database, you can set up a separate repository and
...
 <tr class="even">
 <td><strong>read_write_ldap</strong></td>
-<td><code>org.wso2.carbon.user.core.ldap.ReadWriteLDAPUserStoreManager</code></td>
+<td><code>org.wso2.carbon.user.core.ldap.UniqueIDReadWriteLDAPUserStoreManager</code></td>
 <td><p>Use <code>read_write_ldap</code> for external LDAP user stores to do both read and write operations.</p></td>
 </tr>
 <tr class="odd">
 <td><strong>active_directory</strong></td>
-<td><code>org.wso2.carbon.user.core.ldap.ActiveDirectoryUserStoreManager</code></td>
+<td><code>org.wso2.carbon.user.core.ldap.UniqueIDActiveDirectoryUserStoreManager </code></td>
```
Comment (vithu30) [inline en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-the-primary-user-store.md:37]: Shall we use the relevant mappings used in `infer.json`. For instance, for `UniqueIDActiveDirectoryUserStoreManager` class, userstore name as `active_directory_unique_id`

**Review by vithu30 — APPROVED** [review]

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-the-primary-user-store.md (hunk 3 above)
Comment (Sarangan0219) [reply en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-the-primary-user-store.md:37]: fixed https://github.com/wso2/docs-apim/pull/2174/commits/25962a162efd96371db974af0b1339864f591d11

**Review by Mariangela — APPROVED** [review]

File: en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-read-write-active-directory-user-store.md [hunk 4]
```diff
@@ -68,18 +128,77 @@ Follow the given steps to configure an external Active Directory as the primary
...
-class="org.wso2.carbon.user.core.ldap.ActiveDirectoryUserStoreManager"
-type = "active_directory"
-base_dn = "cn=Users,dc=wso2,dc=org"
-```
-
 !!! note
-    Note that these configurations will automatically applied to the `user-mgt.xml` file so you do not need to edit it.
+    When working with Active Directory;
+
+    -   It is best to enable the `GetAllRolesOfUserEnabled` property in the `AuthorizationManager` as follows. See the documentation on [configuring the Authorization Manager](./../../configuring-the-authorization-manager/) for more information.
```
[the remark this answers is not in the document]
Comment (Mariangela) [reply en/docs/administer/managing-users-and-roles/managing-user-stores/configure-primary-user-store/configuring-a-read-write-active-directory-user-store.md]: @Sarangan0219 , this issue is still pending. Please fix this link.

**Review by Mariangela — APPROVED** [review]

**Pull request merged**